### DISCUSSION TITLE
Injection of ITestOutputHelper in test class

### DISCUSSION THREAD

**Discussion by wolfhermann** [xunit/xunit#2205] [Question] [answered]
Hello xunit team and community,

I wonder if there are other ways to get the instance of the ITestOutputHelper interface in a test class besides the available Constructor injection option.

Background: After updating xUnit in my existing project the established logging via Log4Net does not work within the tests anymore. Therefore a TestOutputHelper instance is now required.

If you use Constructor injection for this purpose, all test classes within my existing project must be adapted accordingly. In this case it would be desirable to get the TestOutputHelper instance directly in a test base class (e.g. via property injection or a static help method of xUnit). An adaptation of the derived test classes would therefore not be necessary.

My question: Does such an approach exist?

Thank you for each answer, a good time and greetings
Wolf

**Comment by bradwilson** [comment, marked as answer]
There is no other way today in v2, without heroic measures.

We are planning support for a test context in v3: https://github.com/xunit/xunit/issues/621

There is an unsupported third party version of this available today for v2. I haven't tried it, but it's spoken well of. https://github.com/SimonCropp/XunitContext

**Reply by wolfhermann** [reply]
Thanks for the quick answer and kind regards!
